vm-bootstorm/azuredeploy.json

```diff
@@ -0,0 +1,366 @@
+      }
+    },
+    "dnsNameForPublicIP": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique Storage Account name to place OS disks."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "East US 2",
+      "allowedValues": [
+        "East US 2",
+        "West US",
+        "Central US",
+        "South Central US",
+        "North Central US",
+        "North Europe"
+      ],
+      "metadata": {
+        "description": "Region to deploy VMs."
+      }
+    },
+    "vmOsSku": {
+      "type": "string",
+      "defaultValue": "2012-R2-Datacenter",
+      "allowedValues": [
+        "2012-R2-Datacenter",
+        "2012-Datacenter",
+        "2008-R2-SP1"
+      ],
+      "metadata": {
+        "description": "The Windows version for the VM. This will pick a fully patched image of this given Windows version. Allowed values: 2008-R2-SP1, 2012-Datacenter, 2012-R2-Datacenter."
+      }
+    },
+    "vmCount": {
+      "type": "int",
+      "defaultValue": 2,
+      "metadata": {
+        "description": "Number of VMs to create."
+      }
+    },
+    "vmAdminPassword": {
+      "type": "securestring",
+      "defaultValue": "vmbootpass!123",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "vmAdminUsername": {
+      "type": "string",
+      "defaultValue": "vmbootadmin",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "vmName": {
+      "type": "string",
+      "defaultValue": "vmbootwin",
+      "metadata": {
+        "description": "Name of the Virtual Machine."
+      }
+    }
+  },
+  "variables": {
+    "imagePublisher": "MicrosoftWindowsServer",
+    "imageOffer": "WindowsServer",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "vmsubnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "vmpublicip",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vmstoraccn",
+    "vmSize": "Standard_A2",
+    "vmOsDiskName": "vmbootosdisk",
+    "vmNicName": "vmnic",
+    "virtualNetworkName": "vmvnet",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "modulesPath": "https://raw.githubusercontent.com/dipakmsft/azure-quickstart-templates/master/vm-bootstorm/",
+    "moduleVMBootAll": "VMBootAll.zip",
+    "modulesUrlVMBootAll": "[concat(variables('modulesPath'),variables('moduleVMBootAll'))]",
+    "configurationFunctionVMBootAll": "VMBootAll.ps1\\ConfigureVMBootAll"
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-06-15",
```

Comment:
Suggest making api version a variable so it's easier to move to newer api versions